### PR TITLE
(GH-1580) Only output log messages when RegularOutput is enabled

### LINKED ISSUE #1580: Logging - warnings for ".registry.bad." files are emitted with "-r" switch
## warnings for corrupted registry files are displayed with the `-r` switch

on each choco.exe call following message is getting emmited (twice)
```
A corrupt .registry file exists at C:\ProgramData\chocolatey\.chocolatey\accurev-client.5.7.1.170613\.registry.bad.
 Open this file in a text editor, and remove/escape any characters that
 are regarded as illegal within XML strings not surrounded by CData.
 These are typically the characters &, `<`, and `>`. Again, this
 is an XML document, so you will see many < and > characters, so just
 focus exclusively in the string values not surrounded by CData. Once
 these have been corrected, rename the .registry.bad file to .registry.
 Once saved, try running the same Chocolatey command that was just
 executed, so verify problem is fixed.
 NOTE: It will not be possible to rename the file in Windows Explorer.
 Instead, you can use the following PowerShell command:
 Move-Item .\.registry.bad .\.registry
```
the choco.exe parameter `-r` should not print warnings:
```
 -r, --limitoutput, --limit-output
     LimitOutput - Limit the output to essential information
```

### PR DESCRIPTION
**Pull request by gep13**
Fixes #1580

### CODE REVIEW

File: src/chocolatey/infrastructure.app/services/ChocolateyPackageInformationService.cs [hunk 1]
```diff
@@ -46,12 +47,12 @@ public ChocolateyPackageInformationService(IFileSystem fileSystem, IRegistryServ
             _filesService = filesService;
         }
 
-        public ChocolateyPackageInformation get_package_information(IPackage package)
+        public ChocolateyPackageInformation get_package_information(IPackage package, ChocolateyConfiguration config)
```
Comment (ferventcoder) [inline src/chocolatey/infrastructure.app/services/ChocolateyPackageInformationService.cs]: This needs an overload

File: src/chocolatey/infrastructure.app/services/IChocolateyPackageInformationService.cs [hunk 2]
```diff
@@ -16,13 +16,14 @@
...
     using NuGet;
 
     public interface IChocolateyPackageInformationService
     {
-        ChocolateyPackageInformation get_package_information(IPackage package);
-        void save_package_information(ChocolateyPackageInformation packageInformation);
-        void remove_package_information(IPackage package);
+        ChocolateyPackageInformation get_package_information(IPackage package, ChocolateyConfiguration config);
+        void save_package_information(ChocolateyPackageInformation packageInformation, ChocolateyConfiguration config);
+        void remove_package_information(IPackage package, ChocolateyConfiguration config);
```
Comment (ferventcoder) [inline src/chocolatey/infrastructure.app/services/IChocolateyPackageInformationService.cs]: All of these need an overload

**Comment by gep13** [comment]
@ferventcoder I have updated based on your comments, and discussion last night.

Let me know if there is additional work required here.  I simply did `new ChocolateyConfiguration` in the overload, wasn't sure if anything needed to be explicitly set here or not.

File: src/chocolatey/infrastructure.app/services/ChocolateyPackageInformationService.cs [hunk 3]
```diff
@@ -47,11 +48,16 @@ public ChocolateyPackageInformationService(IFileSystem fileSystem, IRegistryServ
         }
 
         public ChocolateyPackageInformation get_package_information(IPackage package)
+        {
+            return get_package_information(package, new ChocolateyConfiguration());
```
Comment (ferventcoder) [inline src/chocolatey/infrastructure.app/services/ChocolateyPackageInformationService.cs]: This is probably fine, but I would be a bit more complete by adding the config into the service itself.

Comment (ferventcoder) [reply src/chocolatey/infrastructure.app/services/ChocolateyPackageInformationService.cs]: That way you don't need to pass config around to the functions.

File: src/chocolatey/infrastructure.app/services/IChocolateyPackageInformationService.cs (hunk 2 above)
Comment (ferventcoder) [reply src/chocolatey/infrastructure.app/services/IChocolateyPackageInformationService.cs]: won't need overloads if we don't pass config.

**Comment by gep13** [comment]
@ferventcoder well, that simplified the change quite considerably!

I have left the original whitespace corrections for the files that were originally changed in this PR.  Let me know if you want to back those out as well.

File: src/chocolatey/infrastructure.app/services/ChocolateyPackageInformationService.cs [hunk 4]
```diff
@@ -44,14 +47,23 @@ public ChocolateyPackageInformationService(IFileSystem fileSystem, IRegistryServ
             _fileSystem = fileSystem;
             _registryService = registryService;
             _filesService = filesService;
+            _config = Config.get_configuration_settings();
```
Comment (ferventcoder) [inline src/chocolatey/infrastructure.app/services/ChocolateyPackageInformationService.cs]: 👍

**Pull request merged**